spec/tester_spec.rb

```diff
@@ -24,33 +27,73 @@
     end
   end
 
-  context 'with a custom user dictionary' do
-    it 'scores them against the user dictionary' do
-      result = tester.test('themeforest', ['themeforest'])
+  context "with a custom user dictionary" do
+    it "scores them against the user dictionary" do
+      result = tester.test("themeforest", ["themeforest"])
       expect(result.entropy).to eq 0
       expect(result.score).to eq 0
     end
 
-    it 'matches l33t substitutions on this dictionary' do
-      result = tester.test('th3m3for3st', ['themeforest'])
+    it "matches l33t substitutions on this dictionary" do
+      result = tester.test("th3m3for3st", ["themeforest"])
       expect(result.entropy).to eq 1
       expect(result.score).to eq 0
     end
   end
 
-  context 'with a custom global dictionary' do
-    before { tester.add_word_lists('envato' => ['envato']) }
+  context "with Unicode entries in the password" do
+    it "validates the password" do
+      result = tester.test("✅🐴🔋staple", %w[Theme Forest themeforest])
+      expect(result.entropy).to be_positive
+      expect(result.score).to be_positive
+    end
+  end
+
+  context "with Unicode entries in the dictionary" do
+    it "validates the password" do
+      result = tester.test("correct horse battery staple", %w[✅ 🐴 🔋])
+      expect(result.entropy).to be_positive
+      expect(result.score).to be_positive
+    end
+  end
+
+  context "with Unicode entries in the password and the dictionary" do
+    it "validates the password" do
+      result = tester.test("✅🐴🔋staple", %w[✅ 🐴 🔋])
+      expect(result.entropy).to be_positive
+      expect(result.score).to be_zero
+    end
+  end
+
+  context "with invalid entries in the dictionary" do
+    it "ignores those entries" do
+      result = tester.test("themeforest", [nil, 1, "themeforest"])
+      expect(result.entropy).to eq 0
+      expect(result.score).to eq 0
+    end
+  end
+
+  context "with a custom global dictionary" do
+    before { tester.add_word_lists("envato" => ["envato"]) }
 
-    it 'scores them against the dictionary' do
-      result = tester.test('envato')
+    it "scores them against the dictionary" do
+      result = tester.test("envato")
       expect(result.entropy).to eq 0
       expect(result.score).to eq 0
     end
+
+    context "with invalid entries in a custom dictionary" do
+      before { tester.add_word_lists("themeforest" => [nil, 1, "themeforest"]) }
+
+      it "ignores those entries" do
+        expect(tester.test("themeforest")).to have_attributes(entropy: 0, score: 0, crack_time: 0)
+      end
+    end
   end
 
-  context 'nil password' do
+  context "nil password" do
     specify do
-      expect { tester.test(nil) }.to_not raise_error
```

Comment:
`expect { tester.test(nil) }.to_not raise_error` is one of my personal bugbears!